torch/utils/_cxx_pytree.py

```
@@ -26,6 +28,11 @@
     Union,
 )
 
+import torch
+
+if torch._running_with_deploy():
+    warnings.warn("C++ pytree utilities do not work with torch::deploy.")
+
```

We should probably just raise here instead of warn?